charmtools/charms.py

```diff
@@ -528,6 +551,48 @@ def schema_type(self, **kw):
     )
 
 
+def validate_terms(charm, linter):
+    """Validate terms in charm metadata.
+
+    :param charm: dict of charm metadata parsed from metadata.yaml
+    :param linter: :class:`CharmLinter` object to which info/warning/error
+        messages will be written
+
+    """
+    if 'terms' not in charm:
+        return
+
+    if not isinstance(charm['terms'], list):
+        linter.err('terms: must be a list of term ids')
+
+
+def validate_resources(charm, linter):
+    """Validate resources in charm metadata.
+
+    :param charm: dict of charm metadata parsed from metadata.yaml
+    :param linter: :class:`CharmLinter` object to which info/warning/error
+        messages will be written
+
+    """
+    if 'resources' not in charm:
+        return
+
+    if (not isinstance(charm['resources'], dict) or
+            not charm['resources']):
```

Comment:
Not to play the "but everyone else is doing it" card, but that's the same behavior we've adopted for other keys that require a list or dict. If we want to change this, we'd need to update the validation for all the keys of those types.